### ISSUE TITLE
SPACE_ENTERED/SPACE_ABANDONED fired multiple times for same region

### ISSUE THREAD

**Issue by grzegorzgolec** [kontaktio/kontakt-android-sdk#48]
Iam using sdk 2.0, i need to calculate time which user spent in some region, iam using space events but they are fired multiple times for same region. Iam using 2 test devices galaxy s4 and nexus5, and i have defined 3 regions  (beacon config: tx power set to 2 and 350ms interval). Test device was laying down on the desk during test, distance was about 2.4 meters from nearest beacon (device was all the time in range of the same region but events were fired out of nowhere), below is my scan context config

``` java
final UUID PROXIMITY_UUID = UUID.fromString("f7826da6-4fa2-4e98-8024-bc5b71e08515");

final int major = 30700;
final int[] minors = {30711,30712,30714};


ArrayList<IBeaconRegion> regions = new ArrayList<>();

            for (int i = 0; i< minors.length; i++){
                regions.add(new BeaconRegion(PROXIMITY_UUID, major, minors[i]));
            }

IBeaconScanContext iBeaconScanContext = new IBeaconScanContext.Builder()
                    .setRssiCalculator(RssiCalculators.newLimitedMeanRssiCalculator(5))
                    .setEventTypes(EnumSet.of(EventType.SPACE_ENTERED, EventType.SPACE_ABANDONED)
                    .setDevicesUpdateCallbackInterval(TimeUnit.SECONDS.toMillis(1))
                    .setDistanceSort(DistanceSort.DESC)
                    .setIBeaconRegions(regions)
                    .build();



scanContext = new ScanContext.Builder()
                    .setScanMode(ProximityManager.SCAN_MODE_BALANCED)
                    .setScanPeriod(new ScanPeriod(TimeUnit.SECONDS.toMillis(7), TimeUnit.SECONDS.toMillis(3)))
                    .setForceScanConfiguration(ForceScanConfiguration.DEFAULT)
                    .setActivityCheckConfiguration(ActivityCheckConfiguration.DEFAULT)
                    .setIBeaconScanContext(iBeaconScanContext).build();

```

i have spent half day changeig beacon profile (tx power and interval) or scan peroids of scan context without any good result, on 2 test devices i always got the same result as below

<pre>
Samsung Galaxy S4

09-24 16:56:08.175    4994-4994/? D/BEAR﹕ SCAN READY
09-24 16:56:33.199    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 16:56:33 24/09/2015 
09-24 16:56:43.179    4994-5483/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 16:56:43 24/09/2015 
09-24 16:56:43.449    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 16:56:43 24/09/2015 
09-24 16:57:13.168    4994-5483/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 16:57:13 24/09/2015 
09-24 16:57:18.583    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 16:57:18 24/09/2015 
09-24 16:57:37.832    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Sracz  DATE: 16:57:37 24/09/2015 
09-24 16:57:43.167    4994-5483/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Sracz  DATE: 16:57:43 24/09/2015 
09-24 16:59:19.662    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Sracz  DATE: 16:59:19 24/09/2015 
09-24 16:59:28.170    4994-5483/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 16:59:28 24/09/2015 
09-24 16:59:28.180    4994-5483/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Sracz  DATE: 16:59:28 24/09/2015 
09-24 16:59:28.450    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 16:59:28 24/09/2015 
09-24 16:59:38.170    4994-5483/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 16:59:38 24/09/2015 
09-24 16:59:38.360    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 16:59:38 24/09/2015 
09-24 17:01:28.177    4994-5483/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 17:01:28 24/09/2015 
09-24 17:01:30.549    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 17:01:30 24/09/2015 
09-24 17:03:18.174    4994-5483/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 17:03:18 24/09/2015 
09-24 17:03:19.586    4994-4994/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 17:03:19 24/09/2015 

LG Nexus 5
09-24 17:06:36.953    1505-1505/? D/BEAR﹕ SCAN READY
09-24 17:06:38.202    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 17:06:38 24/09/2015 
09-24 17:06:42.815    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Sracz  DATE: 17:06:42 24/09/2015 
09-24 17:06:51.966    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Sracz  DATE: 17:06:51 24/09/2015 
09-24 17:07:01.965    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 17:07:01 24/09/2015 
09-24 17:07:03.389    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 17:07:03 24/09/2015 
09-24 17:08:20.395    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Sracz  DATE: 17:08:20 24/09/2015 
09-24 17:08:26.965    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Sracz  DATE: 17:08:26 24/09/2015 
09-24 17:08:27.837    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Sracz  DATE: 17:08:27 24/09/2015 
09-24 17:08:36.965    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Sracz  DATE: 17:08:36 24/09/2015 
09-24 17:08:43.086    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Sracz  DATE: 17:08:43 24/09/2015 
09-24 17:08:51.963    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Sracz  DATE: 17:08:51 24/09/2015 
09-24 17:09:08.353    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Sracz  DATE: 17:09:08 24/09/2015 
09-24 17:09:26.965    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Sracz  DATE: 17:09:26 24/09/2015 
09-24 17:09:36.796    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Korytarz  DATE: 17:09:36 24/09/2015 
09-24 17:09:39.525    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Sracz  DATE: 17:09:39 24/09/2015 
09-24 17:09:41.963    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Korytarz  DATE: 17:09:41 24/09/2015 
09-24 17:09:51.697    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Korytarz  DATE: 17:09:51 24/09/2015 
09-24 17:09:51.963    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Sracz  DATE: 17:09:51 24/09/2015 
09-24 17:09:56.965    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Korytarz  DATE: 17:09:56 24/09/2015 
09-24 17:10:01.637    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Korytarz  DATE: 17:10:01 24/09/2015 
09-24 17:10:06.963    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Korytarz  DATE: 17:10:06 24/09/2015 
09-24 17:10:21.966    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 17:10:21 24/09/2015 
09-24 17:10:22.136    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 17:10:22 24/09/2015 
09-24 17:10:36.963    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 17:10:36 24/09/2015 
09-24 17:10:41.656    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 17:10:41 24/09/2015 
09-24 17:10:46.965    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 17:10:46 24/09/2015 
09-24 17:10:47.338    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 17:10:47 24/09/2015 
09-24 17:10:56.965    1505-2137/? D/BEAR﹕ SPACE_ABANDONED REG NAME: Pokój  DATE: 17:10:56 24/09/2015 
09-24 17:10:57.652    1505-1505/? D/BEAR﹕ SPACE_ENTERED REG NAME: Pokój  DATE: 17:10:57 24/09/2015 
</pre>

Is this a right behaviour of android sdk ?  How to overcome this problem ?

**Comment by jakubkontakt** [comment]
That is a bug not intended behaviour of SDK.

We fixed that bug and in the near future will release updated version of SDK.

**Comment by jakubkontakt** [comment]
Please check latest version 2.0.1

**Comment by jakubkontakt** [comment]
Solved in version 2.0.1
